components/cloud/aws/src/s3.rs

```diff
@@ -626,6 +643,67 @@ impl BlobStorage for S3Storage {
     }
 }
 
+struct S3PrefixIter<'cli> {
+    cli: &'cli S3Storage,
+    finished: bool,
+    cont_token: Option<String>,
+    prefix: String,
+}
+
+impl<'cli> S3PrefixIter<'cli> {
+    async fn one_page(&mut self) -> io::Result<Option<Vec<BlobObject>>> {
+        if self.finished {
+            return Ok(None);
+        }
+        let mut input = ListObjectsV2Request::default();
+        input.bucket = String::clone(&self.cli.config.bucket.bucket);
+        input.prefix = Some(self.cli.maybe_prefix_key(&self.prefix));
+        input.continuation_token = self.cont_token.clone();
+        let res = retry_and_count(
+            || self.cli.client.list_objects_v2(input.clone()),
+            "get_one_page",
+        )
+        .await
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+        self.finished = !res.is_truncated.ok_or_else(|| {
+            io::Error::new(io::ErrorKind::InvalidData, "no IsTruncated in response")
```

Comment:
actually seems there is a custom error type already in `CloudError`, can we maybe use that instead?

Reply:
All external storage interfaces should return `io::Error` for now... Also we convert non-io error types of `CouldError` to an `ErrorKind::Other`, I'm afraid that with `could::Error` things won't be better...

Reply:
(update: `cloud::Error` should be neutral over cloud vendors, so I guess `RusotoError` shouldn't be added there... Then I will prefer keep the current impl.)